Docs: Add canvas custom images and icon guidance (#91806)
grafana/grafana: docs/sources/panels-visualizations/visualizations/canvas/index.md

### Icon

The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds or value mappings.

#### Add a custom icon

You can add a custom icon by referencing an SVG file. To add a custom icon, follow these steps:

1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

1. Click **Select**.

> To add a custom icon using another image file type, you must set both the **Icon** and **Background (icon)** image options.

Author: ```suggestion
To add a custom icon using another image file type, you must set both the **Icon > SVG Path** and **Background (icon)** image options.
```

The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds / value mappings.

> You can add custom icons by setting both the **Icon** and **Background (icon)** image options.

Reviewer: ```suggestion
You can customize your icons by setting both the **Icon** SVG path and **Background (icon)** image options.
```

Reviewer: Icons are a bit unique in that they can point to both a custom SVG icon and custom PNG background image. This suggestion makes the language a bit more generic.

Author: My understanding is that you have to set both (based on the support ticket I'm following up on). If I just set a custom SVG icon without setting the background image, then I get nothing. Maybe I'm missing something about the process?

Reviewer: You absolutely can have a custom SVG icon WITHOUT setting a background image. I believe what they mean in that support ticket with, "we should mention that icons consist of 2 layer (icon and background) and that it's required to set a picture as background when we want to have a custom image for icon," is that:

If they have a "picture" (a png) of what they wanted displayed as a canvas icon, they can only render it using the background option as the icon option is only for SVGs. This is because the icon option requires a vector and the background option supports images.

Using an SVG on an icon element:
![Nov-01-2024 16-48-59](https://github.com/user-attachments/assets/b4114a9a-8a0a-4560-8f0b-4b64eb4c6c37)

Using a PNG on an icon element:
![Nov-01-2024 17-00-13](https://github.com/user-attachments/assets/21f93838-9bd8-495c-8d11-10f852d9a0b3)

Using a PNG on a rectangle element:
![Nov-01-2024 17-01-23](https://github.com/user-attachments/assets/ed2d83b3-85ce-47b9-a556-0e13df287fac)

Author: ```suggestion
### Add a custom icon

You can add a custom icon by referencing an SVG file. To add a custom icon, follow these steps:

1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

1. Click **Select**.

To add a custom icon using another image file type, you must set both the **Icon** and **Background (icon)** image options.
```

> The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds / value mappings.

Author: ```suggestion
The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds or value mappings.
```

### Icon

The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds or value mappings.

#### Add a custom icon

You can add a custom icon by referencing an SVG file. To add a custom icon, follow these steps:

1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

1. Click **Select**.

> 1. (Optional) Add a background image to your icon with the **Background (icon)** option by following the steps to [add a custom image](#add-custom-images-to-elements).

Author: ```suggestion
1. (Optional) Add a background image to your icon with the **Background (icon)** option by following the steps to [add a custom image](#add-custom-images-to-elements).

If you don't have an SVG file, you can use a rectangle element instead and reference another image file type. To add a custom image for another element type, follow the steps to [add a custom image](#add-custom-images-to-elements).
```

### Icon

The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds or value mappings.

#### Add a custom icon

You can add a custom icon by referencing an SVG file. To add a custom icon, follow these steps:

1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

1. Click **Select**.
1. (Optional) Add a background image to your icon with the **Background (icon)** option by following the steps to [add a custom image](#add-custom-images-to-elements).

> If you don't have an SVG file, you can use a rectangle element instead an icon and reference another image file type. To add a custom image for another element type, follow the steps to [add a custom image](#add-custom-images-to-elements).

Reviewer: ```suggestion
If you don't have an SVG file, you can use a rectangle element instead of an icon and reference another image file type. To add a custom image for another element type, follow the steps to [add a custom image](#add-custom-images-to-elements).
```

### Icon

The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds or value mappings.

#### Add a custom icon

You can add a custom icon by referencing an SVG file. To add a custom icon, follow these steps:

1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

> 1. Click **Select**.

Author: ```suggestion
1. Click **Select**.
1. (Optional) Add a background image to your icon with the **Background (icon)** option by following the steps to [add a custom image](#add-custom-images-to-elements).
```

### Icon

The icon element lets you add a supported icon to the canvas. Icons can have their color set based on thresholds or value mappings.

#### Add a custom icon

You can add a custom icon by referencing an SVG file. To add a custom icon, follow these steps:

1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

1. Click **Select**.
1. (Optional) Add a background image to your icon with the **Background (icon)** option by following the steps to [add a custom image](#add-custom-images-to-elements).

> If you don't have an SVG file, follow these steps to add a custom icon:
> 
> 1. Under **Icon > SVG Path**, if it's not already selected, select **Fixed** as your file source.
> 1. Change the **Fill color** option to **Transparent**.
> 1. Follow the steps to [add a custom image](#add-custom-images-to-elements) using the **Background (icon)** option.
> 

Author: ```suggestion
```

{{< docs/play title="Canvas Visualization: Buttons" url="https://play.grafana.org/d/c9ea65f5-ed5a-45cf-8fb7-f82af7c3afdf/" >}}

## Add custom images to elements

You can add custom background images to all elements except **Button** by referencing an image URL.

> The image must be hosted on a publicly available URL.

Reviewer: ```suggestion
The image must be hosted at a URL that allows requests from your Grafana instance.
```

To upload a custom image, follow these steps:

1. Under **Background (\<ELEMENT TYPE\>)**, if it's not already selected, select **Fixed** as your image source.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-src-v11.3.png" max-width="300px" alt="Custom image source selection" >}}

1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

> 1. Click **Select**.

Reviewer: This is great!

{{< docs/play title="Canvas Visualization: Buttons" url="https://play.grafana.org/d/c9ea65f5-ed5a-45cf-8fb7-f82af7c3afdf/" >}}

## Add custom images to elements

You can add custom background images to all elements except **Button** by referencing an image URL.
The image must be hosted at a URL that allows requests from your Grafana instance.

To upload a custom image, follow these steps:

1. Under **Background (\<ELEMENT TYPE\>)**, if it's not already selected, select **Fixed** as your image source.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-src-v11.3.png" max-width="300px" alt="Custom image source selection" >}}

1. Click **Select a value** in the field below.
1. In the dialog box that opens, click the **URL** tab.
1. Enter the URL in the field below the **URL** tab.

   {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-canvas-custom-image-v11.3.png" max-width="300px" alt="Add a custom image URL" >}}

> 1. Click **Select**.

Reviewer: Same comment for these. Will it auto increment the list during render?

Author: Yep all good!
